azure-pipelines.nuget.yml

```diff
@@ -0,0 +1,141 @@
+trigger: none
+
+parameters:
+  - name: version
+    displayName: Version to stamp on binaries and nuget package
+    type: string
+
+pool:
+  vmImage: "windows-latest"
+
+variables:
+  solution: "src/AppInstallerCLI.sln"
+  buildConfiguration: "Release"
+  packageName: Microsoft.WindowsPackageManager.Utils
+
+jobs:
+  - job: "Build"
+    variables:
+      BuildVer: $[counter(${{ parameters.version }}, 1)]
+      version: ${{ parameters.version }}.$(BuildVer)
+    steps:
+      - script: echo $(version)
+
+      - task: NuGetToolInstaller@1
+        displayName: Install Nuget
+
+      - task: NuGetCommand@2
+        displayName: Restore Packages
+        inputs:
+          restoreSolution: "$(solution)"
+
+      - task: DotNetCoreCLI@2
+        displayName: DotNet Restore
+        inputs:
+          command: "restore"
+          projects: "**/*.csproj"

```

Comment:
Restoring the solution doesn't restore individual projects?

Reply:
It's a good question, I copied this from https://github.com/microsoft/winget-cli/blob/master/azure-pipelines.yml. I'll try to find out why it's in there

Reply:
We believe it was just an oversight, trying without...

Reply:
Tried only restoring the .sln, that failed. Trying with only csproj

Reply:
Turns out both are necessary. "Dotnet restore" only restores csproj's, not the vcxproj's. The "nuget restore sln" restores both types, but some projects fail at build time with the following:

```
##[error]C:\Program Files\dotnet\sdk\5.0.102\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(241,5): Error NETSDK1004: Assets file 'D:\a\1\s\src\LocalhostWebServer\obj\project.assets.json' not found. Run a NuGet package restore to generate this file.

##[error]C:\Program Files\dotnet\sdk\5.0.102\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(241,5): Error NETSDK1004: Assets file 'D:\a\1\s\src\IndexCreationTool\obj\project.assets.json' not found. Run a NuGet package restore to generate this file.
```

Reply:
Can you add a comment that both are needed?

Reply:
Sure, good idea